pcdsdevices/signal.py

```diff
@@ -13,11 +13,51 @@
 from threading import RLock, Thread
 
 import numpy as np
-from ophyd.signal import Signal
+from ophyd.signal import Signal, EpicsSignalBase, EpicsSignal, EpicsSignalRO
 
 logger = logging.getLogger(__name__)
 
 
+class PytmcEpicsSignal(EpicsSignalBase):
+    """
+    Class for a connection to a pytmc-generated EPICS record.
+
+    This uses the same args as the pragma, so you can refer to the pytmc
+    pragmas to select args for your components. This will automatically append
+    the _RBV suffix and wrap the read/write PVs into the same signal object as
+    appropriate, and pick between a read-only signal and a writable one.
+
+    Under the hood this actually gives you the RW or RO version of the signal
+    depending on your io argument.
+    """
+    def __new__(cls, prefix, *, io, **kwargs):
+        # Same exact check done in pytmc
+        if 'o' in io:
+            return super().__new__(PytmcEpicsSignalRW)
```

Comment:
Interesting! I like the idea.

Couple things:
* Should these pass kwargs through?
* The `io` normalization in pytmc got a bit more [complicated](https://github.com/slaclab/pytmc/blob/8dfb9408c7388bab17563f0a77869c3152759064/pytmc/pragmas.py#L288), but I suppose we could just support the one form here

Reply:
Strangely, when I pass any args/kwargs the thing errors out with something like "object new does not accept kwargs" (paraphrasing from memory). I think I'm doing something slightly wrong that may break later, though this definitely works as-is with the current stack. The result respects `isinstance`, etc. and the child classes follow their inheritance mro's properly.

Would it make sense to make pcdsdevices depend on pytmc and import this `normalize_io` function?